File: src/2974-Minimum-Number-Game/minimum_number_game.cpp
#include <iostream>
#include <vector>
#include <algorithm>

class Solution {
public:
    std::vector<int> numberGame(std::vector<int>& nums) {
        std::vector<int> arr;
        std::sort(nums.begin(), nums.end());
        int i = 0;

        while (i < nums.size()) {
            arr.push_back(nums[i + 1]);
            arr.push_back(nums[i]);
            i += 2;
        }

        return arr;
    }
};

int main() {
    std::vector<int> nums = {5,4,2,3};
    
    Solution solution;

    std::vector<int> output = solution.numberGame(nums);
    
    for (auto it : output) {
        std::cout << it << " ";
    }

    return 0;
}

File: src/2974-Minimum-Number-Game/minimum_number_game.py
from typing import List

class Solution:
    def numberGame(self, nums: List[int]) -> List[int]:
        arr: List[int] = []
        nums.sort()
        i: int = 0

        while i < len(nums):
            arr.append(nums[i+1])
            arr.append(nums[i])
            i += 2

        return arr

if __name__ == "__main__":
    nums: List[int] = [5,4,2,3]
    solution = Solution()

    print(solution.numberGame(nums))